pkg/handlers/defaulthandler.go

```diff
@@ -0,0 +1,235 @@
+	skipArchives bool // Controls whether archive files should be processed.
+}
+
+// configure applies a series of Options to the handler, allowing for flexible configuration of
+// handler-specific settings such as whether to skip binary files. This method supports dynamic
+// customization of handler behavior at runtime.
+func (h *DefaultHandler) configure(opts ...Option) {
+	for _, opt := range opts {
+		opt(h)
+	}
+}
+
+// HandleFile initiates the asynchronous extraction of archive contents. It creates a channel to send extracted data
+// and starts a goroutine that handles the archive opening process. It utilizes a context with a timeout to ensure
+// that the extraction process does not exceed a predefined maximum duration.
+// The function returns a channel that will receive the extracted data bytes and an error if the initial setup fails.
+func (h *DefaultHandler) HandleFile(ctx logContext.Context, input *diskbufferreader.DiskBufferReader) (chan []byte, error) {
+	if h.skipArchives {
+		return nil, nil
+	}
+
+	archiveChan := make(chan []byte, defaultBufferSize)
+	go func() {
+		ctx, cancel := logContext.WithTimeout(ctx, maxTimeout)
+		defer cancel()
+		defer close(archiveChan)
+
+		if err := h.openArchive(ctx, 0, input, archiveChan); err != nil {
+			ctx.Logger().Error(err, "error unarchiving chunk.")
+		}
+	}()
+	return archiveChan, nil
+}
+
+var ErrMaxDepthReached = errors.New("max archive depth reached")
+
+// openArchive recursively extracts content from an archive up to a maximum depth, handling nested archives if necessary.
+// It takes a reader from which it attempts to identify and process the archive format. Depending on the archive type,
+// it either decompresses or extracts the contents directly, sending data to the provided channel.
+// Returns an error if the archive cannot be processed due to issues like exceeding maximum depth or unsupported formats.
+func (h *DefaultHandler) openArchive(ctx logContext.Context, depth int, reader io.Reader, archiveChan chan []byte) error {
+	if common.IsDone(ctx) {
+		return ctx.Err()
+	}
+
+	if depth >= maxDepth {
+		return ErrMaxDepthReached
+	}
+
+	format, arReader, err := archiver.Identify("", reader)
+	if errors.Is(err, archiver.ErrNoMatch) && depth > 0 {
+		return h.handleNonArchiveContent(ctx, arReader, archiveChan)
+	}
+
+	if err != nil {
+		return fmt.Errorf("error identifying archive: %w", err)
+	}
+
+	switch archive := format.(type) {
+	case archiver.Decompressor:
+		// Decompress tha archive and feed the decompressed data back into the archive handler to extract any nested archives.
+		compReader, err := archive.OpenReader(arReader)
+		if err != nil {
+			return err
+		}
+
+		defer compReader.Close()
+
+		return h.openArchive(ctx, depth+1, compReader, archiveChan)
+	case archiver.Extractor:
+		return archive.Extract(logContext.WithValue(ctx, depthKey, depth+1), arReader, nil, h.extractorHandler(archiveChan))
```

Comment:
Isn't `depthKey` an int here? What does that mean as a log key?

Reply:
Correct, `depthKey` is an integer, specifically defined as `ctxKey` to facilitate structured access to values in the context.Context. This definition allows us to use depthKey as a log key to track the current depth of archive processing.

For example within `extractorHandler` we do the following:
```
depth := 0
if ctxDepth, ok := ctx.Value(depthKey).(int); ok {
    depth = ctxDepth
}
```

Reply:
Does this also mean that the log will include weird numeric kvps though?

Reply:
I'm not clear how that works, @mcastorina could you provide some insights?

Reply:
non-string keys are not allowed and will be ignored. This message will print along with the invalid keys

```
non-string key argument passed to logging, ignoring all later arguments
```

Reply:
Just to clarify, when you say _ignored_, you are referring to the key getting displayed in the log message?

Reply:
Yes - the kvp will get added to the context, and it will get added to the logger iff the key is a string.